### PR TITLE
feat(parameter-schema): add new keyword optional for the parameter schema

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #99 

See parent issue for the discussion. The changes a schema are: `If the default value of a parameter is set to anything different from null, the property optional is prohibited`

## Review guidelines

Estimated Time of Review: 5 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

**Review by jjenscodee — APPROVED** [review]

File: schemas/parameters/schema/parameter.schema.json [hunk 1]
```diff
@@ -65,16 +70,39 @@
...
+          }
+        }
+      },
+      "then": {
+        "not": {
+          "required": [
+            "optional"
+          ]
+        }
+      }
```
Comment (eeberhard) [inline schemas/parameters/schema/parameter.schema.json:90]: I'm not certain about the logical implications here; does "not: required" equate to "forbidden"? if you've tested it with VSCode parser and it does what you expect, then that's great. But in my head it makes more sense if you phrase it the other way around: don't include the `optional` property to begin with, but if default value is null, then include `optional`and make it required.

```suggestion
      "if": {
        "properties": {
          "default_value": {
            "const": null
          }
        }
      },
      "then": {
        "properties": {
          "optional": {
            "description": "Specify if this parameter if optional if the default value is null",
            "type": "boolean"
          },
          "required": [
            "optional"
          ]
        }
      }
```

Comment (domire8) [reply schemas/parameters/schema/parameter.schema.json:90]: As explained in person, I believe that `not: required` actually means forbidden, especially with the VSCode parser.

File: schemas/parameters/schema/parameter.schema.json [hunk 2]
```diff
@@ -48,6 +49,10 @@
         }
       ]
     },
+    "optional": {
+      "description": "Specify if this parameter if optional if the default value is null",
```
Comment (eeberhard) [inline schemas/parameters/schema/parameter.schema.json:53]: nitpick for wording and clarity
```suggestion
      "description": "Specify if this parameter is optional (only applicable if the default value is null)",
```

**Review by eeberhard — APPROVED** [review]
Small nitpick, then looks good to me 👍 hope the workflow to generate new PR from this version change is successful!

**Pull request merged**